java-frontend/src/main/java/org/sonar/java/se/ExplodedGraphWalker.java

```diff
@@ -793,7 +799,14 @@ private void executeVariable(VariableTree variableTree, @Nullable Tree terminato
       if (terminator != null && terminator.is(Tree.Kind.FOR_EACH_STATEMENT)) {
         sv = constraintManager.createSymbolicValue(variableTree);
       } else if (variableTree.parent().is(Tree.Kind.CATCH)) {
-        sv = getCaughtException(variableTree.symbol().type());
+        SymbolicValue thrownSV = getCaughtException(variableTree.symbol().type());
+        // replace the exceptional sv by a sv wrapping it, because the exception will be used as a variable within the block
+        sv = new SymbolicValue() {
```

Comment:
I think it will be better to have dedicated `static` class for this type of symbolic value, to avoid capturing implicit reference to the outer `ExplodedGraphWalker` . Also I would prefer not to override `wrappedValue` but provide specific method for this purpose.  You should use `constraintManager` and symbolic value factory mechanism to create instance of SV.

Reply:
I'll rethink a bit that, indeed.